netlogo-core/src/main/lab/StatsExporter.scala

```diff
@@ -0,0 +1,240 @@
+            val n = handleNonNumeric(noQuotes, -1)
+            if (n.isNaN) params += noQuotes
+            else params += n
+          }
+        }}
+        val step = measurements(0).toInt
+        val paramsList = params.drop(1).toList
+        if (!data.contains(paramsList)) {
+          data(paramsList) = new HashMap[Int, Measurements]()
+        }
+        if (!data(paramsList).contains(step)) {
+          data(paramsList)(step) = new ListBuffer[List[Double]]()
+        }
+        data(paramsList)(step) += measurements.drop(1).toList
+      }
+      row += 1
+    }
+    bufferedReader.close()
+    data
+  }
+
+  private def extractFromSpreadsheet(fileName: String): Data = {
+    val bufferedReader: BufferedReader = new BufferedReader(new FileReader(fileName))
+    var line = ""
+    val data = new HashMap[List[Any], DataPerStep]() // use a hashmap because we don't know how many runs there are
+    var reachedRunData = false
+    while ({line = bufferedReader.readLine; line} != null) {
+      val rowElements = line.split("[,\n]")
+
+      // Read lines until the line that contains "[all run data]" or "[initial & final values]" is reached
+      if (reachedRunData) {
+        var runNumber = 1
+
+        // Data is stored along the row, so increment our current position by the number of metrics (+1 to account for the steps column)
+        // to access each run
+        for (i <- 1 to rowElements.length - 1 by protocol.metrics.length + 1) {
+          val params = {
+            if (protocol.sequentialRunOrder) {
+              paramCombinations((runNumber - 1) / protocol.repetitions)
+            } else {
+              paramCombinations((runNumber - 1) % protocol.repetitions)
+            }
+          }
+          val step = rowElements(i).split("\"")(1).toInt
+          val measurements = rowElements.slice(i+1, protocol.metrics.length + i + 1).zipWithIndex.map{case (entry, col) => {
+            val noQuotes = entry.split("\"")(1)
+            handleNonNumeric(noQuotes, col)
+          }}.toList
+
+          if (!data.contains(params)) {
+            data(params) = new HashMap[Int, Measurements]()
+          }
+          if (!data(params).contains(step)) {
+            data(params)(step) = new ListBuffer[List[Double]]()
+          }
+          data(params)(step) += measurements
+          runNumber += 1
+        }
+      } else if (rowElements.length > 0  && ((rowElements(0) equals "\"[all run data]\"") || (rowElements(0) equals "\"[initial & final values]\""))) {
+        reachedRunData = true
+      }
+    }
+    data
+  }
+
+  // private def extractFromSpreadsheet(spreadsheet: SpreadsheetExporter): Data = {
+  //   val data = new HashMap[List[Any], DataPerStep]()
+  //   val spreadsheetData = spreadsheet.runs
+  //   for ((runNumber, run) <- spreadsheetData) {
+  //     val params = run.settings.map(_._2)
+  //     if (!data.contains(params)) {
+  //       data(params) = new HashMap[Int, Measurements]()
+  //     }
```

Comment:
don't leave in commented out code. save it separately if you want it for some reason